File: Backend/backend-sgac/src/main/java/org/uteq/sgacfinal/entity/TipoEstadoEvidencia.java
package org.uteq.sgacfinal.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "tipo_estado_evidencia", schema = "ayudantia")
public class TipoEstadoEvidencia {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_tipo_estado_evidencia", nullable = false)
    private Integer id;

    @Size(max = 50)
    @NotNull
    @Column(name = "nombre_estado", nullable = false, length = 50)
    private String nombreEstado;

    @Column(name = "descripcion", length = Integer.MAX_VALUE)
    private String descripcion;

    @Column(name = "activo")
    private Boolean activo;

}

File: Frontend/frontend-sgac/src/app/core/models/catalogos/TipoEstadoEvidencia.ts
export interface TipoEstadoEvidenciaResponse {
  id: number;
  nombre_estado: string;
  descripcion: string;
  codigo: string;
  activo: boolean;
}

export interface TipoEstadoEvidenciaRequest {
  nombreEstado: string;
  descripcion: string;
  codigo: string;
}
